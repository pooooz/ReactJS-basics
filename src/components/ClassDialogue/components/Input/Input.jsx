import React from 'react';
import styles from './Input.module.scss';

export class Input extends React.Component {
  render() {
    return (
      <>
        <div className={styles.input_wrap}>
          <input
            className={styles.input_green}
            type="text"
            value={this.props.value}
            onChange={this.props.change}
          />
        </div>
      </>
    );
  }
}
