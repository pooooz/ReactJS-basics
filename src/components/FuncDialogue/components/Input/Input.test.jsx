import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Input } from './Input';

describe('FuncInput', () => {
  it('Render component', () => {
    render(<Input />);
  });

  it('Render with snapshot', () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Multiple render', () => {
    render(
      <>
        <Input />
        <Input />
      </>
    );
    expect(screen.getAllByRole('textbox').length).toBe(2);
  });

  it('Render with value and change handler', () => {
    render(<Input />);
    fireEvent.input(screen.getByRole('textbox'), {
      target: { value: 'message' },
    });
    expect(screen.getByDisplayValue(/message/)).toBeInTheDocument();
  });
});
