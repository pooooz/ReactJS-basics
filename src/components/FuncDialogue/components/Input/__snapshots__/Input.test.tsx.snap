// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FuncInput Render with snapshot 1`] = `
<DocumentFragment>
  <div>
    <input
      type="text"
      value="Test"
    />
  </div>
</DocumentFragment>
`;
